File: My Solutions/2022/Day 3/1.py
f = open("in.txt", "r")
def ord2(char):
    if char.islower():
        return ord(char) - 96
    else:
        return ord(char) - 38
total = 0
for line in f.read().split("\n"):
    brk = False
    rucka = line[slice(0,len(line)//2)]
    ruckb = line[slice(len(line)//2,len(line))]
    for char in rucka:
        for chbr in ruckb:
            if char == chbr:
                total += ord2(char)
                brk = True
                break
        if brk:
            break
print(total)

File: My Solutions/2023/Day 1/1.js
const fs = require("fs");

var text = fs.readFileSync("./in.txt").toString();
var total = 0;
for (let line of text.split("\n")) {
    let thisNum = "";
    for (let char of line) {
        if (!isNaN(Number(char)) && !/\s/.test(char)) {
            if (thisNum.length < 2) {
                thisNum += char;
            } else {
                thisNum = thisNum.slice(0,-1) + char;
            }
        }
    }
    if (thisNum.length == 1) {
        thisNum += thisNum;
    }
    total += Number(thisNum);
}
console.log(total);